cpp/include/cudf/detail/gather.cuh

```diff
@@ -50,24 +51,24 @@ struct bounds_checker {
 template <bool ignore_out_of_bounds, typename MapIterator>
 __global__ void gather_bitmask_kernel(table_device_view source_table,
                                       MapIterator gather_map,
-                                      mutable_table_device_view destination_table,
+                                      bitmask_type * masks[],
+                                      size_type destination_table_num_rows,
                                       size_type* valid_counts) {
 
   for (size_type i = 0; i < source_table.num_columns(); i++) {
 
     constexpr int warp_size = 32;
 
     column_device_view source_col = source_table.column(i);
-    mutable_column_device_view destination_col = destination_table.column(i);
 
     if (source_col.has_nulls()) {
```

Comment:
Will `masks[i]` be set to nullptr if a given destination column is not nullable? I want to make sure this change is compatible with https://github.com/rapidsai/cudf/pull/3354/files#diff-1511263c1493ba98239e38a46ebf1d8a

Reply:
Yes, https://github.com/rapidsai/cudf/blob/703221ad8e13593d5311146e6386964b84966233/cpp/include/cudf/detail/gather.cuh#L335